### PR TITLE
Add MarginalNormal and CovarianceMatrix to normal

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
Wow, this diff looks a lot bigger than it is. What happened was adding MarginalNormal, CovarianceMatrix, and I moved ConditionalNormal to preserve alphabetical ordering.

**Comment by kortschak** [comment]
Why do we need `CovarianceMatrix`? Presumably the sigma is available from during construction - is it for cases where the user has called `ConditionNormal`? If so, can't they do `ss.SubsetSym(sigma, vars)`? If it's just for convenience, I'm OK with that, but it would be nice to know.

**Comment by btracey** [comment]
One reason is that you could be given the Normal (or a list of normal) rather than constructing it yourself and so may not have the Sigma. If you have computed MarginalNormal you can use the call you mention, but that is not the correct update for ConditionNormal. To find Sigma you'd have to do the whole update rule yourself. The final reason is that CovarianceMatrix is a more general operation that could easily have a reasonable interface with it (just like StdDev can be computed for many different distributions).

**Comment by kortschak** [comment]
LGTM but please wait for @vladimir-ch.

File: distmv/normal.go [hunk 1]
```diff
@@ -221,6 +156,106 @@ func (n *Normal) ConditionNormal(observed []int, values []float64, src *rand.Ran
...
+	floats.SubTo(xMinusMu, x, n.mu)
+	d := mat64.NewVector(dim, xMinusMu)
+	tmp := make([]float64, dim)
+	tmpVec := mat64.NewVector(dim, tmp)
+	tmpVec.SolveCholeskyVec(&n.chol, d)
+	return c - 0.5*floats.Dot(tmp, xMinusMu)
+}
+
+// MarginalNormal returns the marginal distribution of the given input variables.
+// That is, MarginalNorm returns
```
Comment (vladimir-ch) [inline distmv/normal.go]: MarginalNormal

**Comment by vladimir-ch** [comment]
I'm wondering if `ConditionNormal` shouldn't be called `ConditionalNormal` after all for consistency with `MarginalNormal`.

LGTM

**Comment by btracey** [comment]
I think no, but I don't feel strongly. Marginal keeps the variables that
are input (it returns the marginal distribution of those variables).
Condition removes the input variables, it conditions them out

**Pull request merged**